.gitlab-ci.yml

```diff
@@ -0,0 +1,35 @@
+image: espressomd/buildenv-espresso-ubuntu-cuda:python
+build_and_test:
+  stage: build
+  script:
+    - bash maintainer/cuda_build.sh
+
+
+sphinx_documentation:
+  stage: deploy
+  only:
+    - doc
+  before_script:
+    # Install ssh-agent if not already installed, it is required by Docker.
+    # (change apt-get to yum if you use a CentOS-based image)
+    - 'which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )'
+
+    # Run ssh-agent (inside the build environment)
+    - eval $(ssh-agent -s)
+
+    # Add the SSH key stored in SSH_PRIVATE_KEY variable to the agent store
+    - ssh-add <(echo "$SSH_PRIVATE_KEY")
+
+    # For Docker builds disable host key checking. Be aware that by adding that
+    # you are suspectible to man-in-the-middle attacks.
+    # WARNING: Use this only with the Docker executor, if you use it with shell
+    # you will overwrite your user's SSH config.
```

Comment:
There is a safety check (`[[ -f /.dockerenv ]]`) that prevents just that scenario.